internal/plugin/connectors/http/generic/oauth/v1/protocol_test.go

```diff
@@ -0,0 +1,352 @@
+package oauth1protocol
+
+import (
+	"io"
+	"io/ioutil"
+	gohttp "net/http"
+	"net/url"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func Test_extractOAuthParams(t *testing.T) {
+	paramMap := map[string]string{
+		consumerKey:          "consumerKey",
+		consumerSecret:       "consumerSecret",
+		token:                "token",
+		tokenSecret:          "tokenSecret",
+		oauthNonce:           "oauth_nonce",
+		oauthSignatureMethod: "oauth_signature_method",
+		oauthVersion:         "oauth_version",
+		oauthTimestamp:       "oauth_timestamp",
+	}
+	result := extractOAuthParams(paramMap)
+	// Nonce and TimeStamp are generated at runtime and unknown
+	result.TimeStamp = ""
+	result.Nonce = ""
+
+	want := oAuth1{
+		ConsumerKey:     "consumerKey",
+		ConsumerSecret:  "consumerSecret",
+		Nonce:           "",
+		SignatureMethod: "HMAC-SHA1",
+		TimeStamp:       "",
+		Token:           "token",
+		TokenSecret:     "tokenSecret",
+		Version:         "1.0",
+	}
+	if reflect.DeepEqual(result, want) {
+		t.Log("extractOAuthParams() Passed!")
```

Comment:
Throughout this file, you use standard logging and error throwing, which can be difficult to read when running tests, and isn't the standard for testing in secretless. IIRC the logging / error style is suggested by Goland, but you can ignore that.

I'd refer to `generic/config-test.go` to see how we use the `assert` collection. It's very similar to what you have here, and easy to implement. You shouldn't have to write any custom logging formats, just provide the expected and actual values.